### PR TITLE
[Refactor] support select/insert/delete on computeNodes and run in default warehouse

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java [hunk 1]
```diff
@@ -224,8 +224,10 @@ private boolean handleHbResponse(HeartbeatResponse response, boolean isReplay) {
             case BACKEND: {
                 BackendHbResponse hbResponse = (BackendHbResponse) response;
                 ComputeNode computeNode = nodeMgr.getBackend(hbResponse.getBeId());
+                boolean isBackend = true;
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/system/HeartbeatMgr.java:227]: not necessary?

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 2]
```diff
@@ -470,12 +470,12 @@ public Set<Long> getBackendIdsByShard(long shardId) throws UserException {
                         //  saveImage(). Refer to: https://starrocks.atlassian.net/browse/SR-16340
                         if (workerInfo.getWorkerPropertiesMap().containsKey("be_port")) {
                             int bePort = Integer.parseInt(workerInfo.getWorkerPropertiesMap().get("be_port"));
-                            Backend be = GlobalStateMgr.getCurrentSystemInfo()
-                                    .getBackendWithBePort(pair[0], bePort);
-                            if (be == null) {
+                            ComputeNode cn = GlobalStateMgr.getCurrentSystemInfo()
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:473]: what if the backend id is a BE not a CN?

File: fe/fe-core/src/main/java/com/starrocks/lake/delete/LakeDeleteJob.java [hunk 3]
```diff
@@ -102,9 +103,15 @@ public void run(DeleteStmt stmt, Database db, Table table, List<Partition> parti
                     beToTablets.size());
             SystemInfoService systemInfoService = GlobalStateMgr.getCurrentSystemInfo();
             for (Map.Entry<Long, List<Long>> entry : beToTablets.entrySet()) {
-                Backend backend = systemInfoService.getBackend(entry.getKey());
+                // TODO: need to refactor after be split into cn + dn
+                ComputeNode backend = systemInfoService.getBackend(entry.getKey());
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/delete/LakeDeleteJob.java:107]: can we add new interface to SystemInfoService, called `getComputeNodeExtended()` which search backend first and then if shared_data mode, search from compute node as well. 

this way when we done the be split. we only change this interface, not all the places around.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 67 (61.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|0|4|00.00%|[487, 488, 489, 490]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|1|00.00%|[249]|
|:large_blue_circle:|com/starrocks/qe/SimpleScheduler.java|2|8|25.00%|[91, 105, 146, 147, 148, 150]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|6|50.00%|[350, 1095, 1096]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|12|50.00%|[534, 535, 536, 537, 538, 539]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|2|50.00%|[105]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|1|2|50.00%|[108]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|7|9|77.78%|[434, 522]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|5|6|83.33%|[118]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|8|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 4]
```diff
@@ -525,10 +527,22 @@ public SystemInfoService getOrCreateSystemInfo(Integer clusterId) {
     public TNodesInfo createNodesInfo(Integer clusterId) {
         TNodesInfo nodesInfo = new TNodesInfo();
         SystemInfoService systemInfoService = getOrCreateSystemInfo(clusterId);
-        for (Long id : systemInfoService.getBackendIds(false)) {
-            Backend backend = systemInfoService.getBackend(id);
-            nodesInfo.addToNodes(new TNodeInfo(backend.getId(), 0, backend.getHost(), backend.getBrpcPort()));
+        // use default warehouse
+        Warehouse warehouse = warehouseMgr.getDefaultWarehouse();
+        // TODO: need to refactor after be split into cn + dn
+        if (warehouse != null && RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:533]: suppose after the default warehouse change, there will be no `warehouse == null` case?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 67 (61.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|0|4|00.00%|[486, 487, 488, 489]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|1|00.00%|[249]|
|:large_blue_circle:|com/starrocks/qe/SimpleScheduler.java|2|8|25.00%|[91, 105, 146, 147, 148, 150]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|6|50.00%|[350, 1095, 1096]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|12|50.00%|[534, 535, 536, 537, 538, 539]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|2|50.00%|[105]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|1|2|50.00%|[108]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|7|9|77.78%|[434, 522]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|5|6|83.33%|[118]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|8|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 5]
```diff
@@ -470,12 +470,16 @@ public Set<Long> getBackendIdsByShard(long shardId) throws UserException {
...
                                     .getBackendWithBePort(pair[0], bePort);
-                            if (be == null) {
-                                LOG.warn("can't find backendId with bePort:{} for {}.", bePort, workerAddr);
+                            if (cn == null) {
+                                cn = GlobalStateMgr.getCurrentSystemInfo()
+                                        .getComputeNodeWithBePort(pair[0], bePort);
+                                if (cn == null) {
+                                    LOG.warn("can't find backendId with bePort:{} for {}.", bePort, workerAddr);
+                                }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:480]: else { backendId = cn.getId(); }

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java [hunk 6]
```diff
@@ -95,7 +96,9 @@ public Set<Long> getBackendIds() {
             return Collections.emptySet();
         }
         try {
-            return GlobalStateMgr.getCurrentStarOSAgent().getBackendIdsByShard(getShardId());
+            Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().getDefaultWarehouse();
+            long workerGroupId = warehouse.getAnyAvailableCluster().getWorkerGroupId();
```
[the remark this answers is not in the document]
Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java:100]: ??

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 41 / 71 (57.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|0|4|00.00%|[486, 487, 488, 489]|
|:large_blue_circle:|com/starrocks/warehouse/Cluster.java|0|1|00.00%|[78]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|0|1|00.00%|[249]|
|:large_blue_circle:|com/starrocks/qe/SimpleScheduler.java|2|8|25.00%|[91, 105, 146, 147, 148, 150]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|3|6|50.00%|[350, 1095, 1096]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|12|50.00%|[534, 535, 536, 537, 538, 539]|
|:large_blue_circle:|com/starrocks/lake/LakeTablet.java|3|6|50.00%|[84, 85, 86]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|2|50.00%|[105]|
|:large_blue_circle:|com/starrocks/lake/delete/LakeDeleteJob.java|1|2|50.00%|[108]|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|7|9|77.78%|[434, 522]|
|:large_blue_circle:|com/starrocks/lake/Utils.java|5|6|83.33%|[118]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|8|9|88.89%|[481]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/planner/OlapScanNode.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/ShardDeleter.java|1|1|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**